Handle EntryNotFoundError when retrieving Illumina sequencing run by device internal ID

karlnyr opened the pull request:
## Description

Fixes bug in https://github.com/Clinical-Genomics/cg/issues/3555

### Added

- Error handler when we do not have an entry in statusdb

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

ChrOertlin commented on cg/services/illumina/post_processing/post_processing_service.py:
```diff
@@ -208,12 +209,14 @@ def post_process_illumina_flow_cell(
         has_backup: bool = False
-        if sequencing_run:
+        try:
+            sequencing_run: IlluminaSequencingRun | None = (
```
```suggestion
            sequencing_run: IlluminaSequencingRun = (
```

ChrOertlin commented on cg/services/illumina/post_processing/post_processing_service.py:
```diff
@@ -208,12 +209,14 @@ def post_process_illumina_flow_cell(
+                self.status_db.get_illumina_sequencing_run_by_device_internal_id(run_directory_data.id)
+            )
             has_backup: bool = sequencing_run.has_backup
+        except EntryNotFoundError as e:
```
```suggestion
        except EntryNotFoundError as error:
```

ChrOertlin reviewed (approved):
Missed this one, good catch

islean reviewed (approved):
Looks good!

karlnyr commented:
Testing in stage:

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/127c1ca2-dd71-4761-9cbe-1c63cec13b5d">

`HK33MDRX3` not in database.

When running post processing

```Command
[13:14] [hiseq.clinical@hasta:/home/proj/stage/sequencing_data/illumina/demultiplexed-runs/230912_A00187_1009_AHK33MDRX3] [S_base]  $ cg demultiplex finish illumina-run 230912_A00187_1009_AHK33MDRX3
Running cg demultiplex.
Post-process Illumina run 230912_A00187_1009_AHK33MDRX3
Samplesheet passed validation
Return file: 6502244
Deleting file /home/proj/stage/housekeeper-bundles/ACC12637A1/2023-09-11/HK33MDRX3_ACC12637A1_S3_L001_R1_001.fastq.gz from disc
Deleting file 6502244 from housekeeper
.........
Linked file: /home/proj/stage/sequencing_data/illumina/demultiplexed-runs/230912_A00187_1009_AHK33MDRX3/HK33MDRX3_ACC12637A6_S15_L002_R2_001.fastq.gz -> /home/proj/stage/housekeeper-bundles/ACC12637A6/2023-09-11/HK33MDRX3_ACC12637A6_S15_L002_R2_001.fastq.gz
File added to Housekeeper bundle ACC12637A6
Samplesheet passed validation
Bundle HK33MDRX3 already has a file with the same name as /home/proj/stage/sequencing_data/illumina/sequencing-runs/230912_A00187_1009_AHK33MDRX3/HK33MDRX3_demultiplex.stdout
Added demux log file /home/proj/stage/sequencing_data/illumina/sequencing-runs/230912_A00187_1009_AHK33MDRX3/HK33MDRX3_demultiplex.stdout to Housekeeper.
Bundle HK33MDRX3 already has a file with the same name as /home/proj/stage/sequencing_data/illumina/sequencing-runs/230912_A00187_1009_AHK33MDRX3/HK33MDRX3_demultiplex.stderr
Added demux log file /home/proj/stage/sequencing_data/illumina/sequencing-runs/230912_A00187_1009_AHK33MDRX3/HK33MDRX3_demultiplex.stderr to Housekeeper.
Bundle HK33MDRX3 already has a file with the same name as /home/proj/stage/sequencing_data/illumina/sequencing-runs/230912_A00187_1009_AHK33MDRX3/RunParameters.xml
Added run parameters file /home/proj/stage/sequencing_data/illumina/sequencing-runs/230912_A00187_1009_AHK33MDRX3/RunParameters.xml to HK33MDRX3 in Housekeeper.
```

And it is parsed and it appears in the database:

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/b498159e-dfb5-49c5-9677-2622170c84d7">
